mmhuman3d/models/heads/expose_head.py

```diff
@@ -0,0 +1,488 @@
+            self.pose_param_decoders['{}_decoder'.format(pose_name)] = pose_decoder
+            pose_dim = pose_decoder.get_dim_size()
+            pose_mean = pose_decoder.get_mean()
+            if pose_param['rotate_axis_x']:
+                pose_mean[3] = -1
+            idxs = list(range(start, start + pose_dim))
+            idxs = torch.tensor(idxs, dtype=torch.long)
+            self.register_buffer('{}_idxs'.format(pose_name), idxs)
+            start += pose_dim
+            mean_lst.append(pose_mean.view(-1))
+        return start, mean_lst
+
+    def get_camera_param(self, camera_cfg):
+        camera_pos_scale = camera_cfg.get('pos_func')
+        if camera_pos_scale == 'softplus':
+            camera_scale_func = F.softplus
+        elif camera_pos_scale == 'exp':
+            camera_scale_func = torch.exp
+        elif camera_pos_scale == 'none' or camera_pos_scale == 'None':
+            def func(x):
+                return x
+            camera_scale_func = func
+        mean_scale = camera_cfg.get('mean_scale', 0.9)
+        if camera_pos_scale == 'softplus':
+            mean_scale = np.log(np.exp(mean_scale) - 1)
+        elif camera_pos_scale == 'exp':
+            mean_scale = np.log(mean_scale)
+        camera_mean = torch.tensor([mean_scale, 0.0, 0.0], dtype=torch.float32)
+        camera_param_dim = 3
+        return camera_mean, camera_param_dim, camera_scale_func
+
+    def flat_params_to_dict(self, param_tensor):
+        smplx_dict = {}
+        raw_dict = {}
+        for pose_param in self.pose_param_conf:
+            pose_name = pose_param['name']
+            pose_idxs = getattr(self,f'{pose_name}_idxs')
+            decoder = self.pose_param_decoders[f'{pose_name}_decoder']
+            pose = torch.index_select(param_tensor, 1 , pose_idxs)
+            raw_dict[f'raw_{pose_name}'] = pose.clone()
+            smplx_dict[pose_name] = decoder(pose)
+        return smplx_dict, raw_dict
+
+    def get_mean(self, name, batch_size):
+        mean_param = self.regressor.get_mean().view(-1)
+        if name is None:
+            return mean_param.reshape(1,-1).expand(batch_size,-1)
+        idxs = getattr(self,f'{name}_idxs')
+        return mean_param[idxs].reshape(1,-1).expand(batch_size, -1)
+
+    def get_num_betas(self):
+        return self.num_betas
+
+    def get_num_expression_coeffs(self):
+        return self.num_expression_coeffs
+
+    @abstractmethod
+    def forward(self, features):
+        pass
+
+    
+class ExPoseBodyHead(ExPoseHead):
+    def __init__(self, init_cfg=None, num_betas: int = 10, num_expression_coeffs: int = 10, mean_pose_path: str = '', shape_mean_path: str = '', input_feat_dim: int = 2048,  regressor_cfg: dict = None, camera_cfg: dict = None):
+        super().__init__(init_cfg)
+        self.num_betas = num_betas
+        self.num_expression_coeffs = num_expression_coeffs
+        # poses
+        self.pose_param_conf = [
```

Comment:
We probably want to control this parameters in config file

Reply:
Controllable now.